zapread.com/Controllers/MessagesController.cs

```diff
@@ -873,18 +873,30 @@ public async Task<JsonResult> SendMessage(int id, string content, bool? isChat)
         }
 
         [HttpPost]
-        [AllowAnonymous]
         public JsonResult GetMessage(int id, int userId)
         {
+            var localUserId = User.Identity.GetUserId();
+
+            if(localUserId == null)
+            {
+                return Json(new { success = false, result = "Failure", message = "Error receiving message." });
+            }
+
             string HTMLString = "";
 
             using (var db = new ZapContext())
             {
                 var msg = db.Messages
                     .Include("From")
                     .Include("To")
+                    .Where(m => m.From.AppId == localUserId || m.To.AppId == localUserId)
                     .SingleOrDefault(m => m.Id == id);
 
+                if (msg == null)
+                {
+                    return Json(new { success = false, result = "Failure", message = "Error receiving message." });
```

Comment:
`Response.StatusCode = (int)HttpStatusCode.BadRequest;`

Reply:
Edited, not tested yet as I'm not at home right now.